CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.9)
+cmake_minimum_required(VERSION 3.12)
```

Comment:
This is higher than what both VecGeom (3.1) and ROOT (3.9) require, so I'm a bit reluctant to increase. However, if it's really necessary, we can consider updating all three at once. Since VecCore is a dependency for ROOT and VecGeom, I'd like to avoid having higher requirements for VecCore. We considered requiring 3.12 for ROOT back in the day, but decided to stick with 3.9 due to lack of newer CMake packaged in older OSs supported by ROOT.